src/controller.rs

```diff
@@ -67,13 +67,18 @@ impl<'b> Controller<'b> {
         }
 
         let attached_to_pager = output_type.is_pager();
-        let rw_cycle_detected = !attached_to_pager && {
-            let identifiers: Vec<_> = inputs
-                .iter()
-                .flat_map(clircle::Identifier::try_from)
-                .collect();
-            clircle::stdout_among_inputs(&identifiers)
+        let stdout_identifier = if !attached_to_pager {
+            clircle::Identifier::try_from(clircle::Stdio::Stdout).ok()
+        } else {
+            None
```

Comment:
Maybe reverse the two values and use the positive `if attached_to_pager { … }`

Reply:
Agreed.

Rust 1.50 would have had our back here with `(!attached_to_pager).then(clircle::Identifier::stdout)` (I will probably add the `Identifier::stdout` anyways) :laughing:

Reply:
Oh, that's a nice helper function :+1: